src/ios/DualCameraSessionManager.swift

```diff
@@ -0,0 +1,371 @@
+        queue.async { [weak self] in
+            guard let self = self else { 
+                DispatchQueue.main.async { completion(false) }
+                return 
+            }
+
+            self.isConfiguring = true
+            self.session.beginConfiguration()
+            var setupSuccess = true
+
+            defer { 
+                self.session.commitConfiguration()
+                self.isConfiguring = false
+                self.isSetupComplete = setupSuccess
+                DispatchQueue.main.async {
+                    completion(setupSuccess)
+                }
+            }
+
+            setupSuccess = self.setupBackCamera() && 
+                          self.setupFrontCamera() && 
+                          self.setupMicrophone()
+
+            if setupSuccess {
+                self.backOutput.setSampleBufferDelegate(self, queue: self.queue)
+                self.frontOutput.setSampleBufferDelegate(self, queue: self.queue)
+                if let audioOutput = self.audioOutput {
+                    audioOutput.setSampleBufferDelegate(self, queue: self.queue)
+                }
+            }
+        }
+    }
+
+    func startRecording(with recorder: VideoRecorder) {
+        queue.async { [weak self] in
+            guard let self = self else { return }
+
+            guard self.isReady() && !self.isRecording else {
+                print("Cannot start recording: session not ready or already recording")
+                return
+            }
+
+            self.videoRecorder = recorder
+            self.isRecording = true
+            self.videoMixer.lockOrientation()
+
+            let isLandscape = UIDevice.current.orientation.isLandscape
+            if isLandscape {
+                self.videoMixer.pipFrame = CGRect(x: 0.03, y: 0.03, width: 0.25, height: 0.25)
+            } else {
+                self.videoMixer.pipFrame = CGRect(x: 0.05, y: 0.05, width: 0.3, height: 0.3)
+            }
+        }
+    }
+    
+    func stopRecording() {
+        queue.async { [weak self] in
+            guard let self = self else { return }
+            self.videoMixer.unlockOrientation()
+            self.videoRecorder = nil
+            self.isRecording = false
+        }
+    }
+
+    func startSession() {
+        queue.async { [weak self] in
+            guard let self = self else { return }
+
+            if self.isSetupComplete && !self.isConfiguring && !self.session.isRunning {
+                self.session.startRunning()
+            }
+        }
+    }
+    
+    func isReady() -> Bool {
+        stateLock.lock()
+        defer { stateLock.unlock() }
+        return _isSetupComplete && !_isConfiguring
+    }
+
+    func stopSession() {
+        queue.async { [weak self] in
+            guard let self = self else { return }
+      
+            if self.isConfiguring {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+                    self.stopSession()
+                }
```

Comment:
if isConfiguring stays true, wouldn't this end up in an infinite loop? maybe add a retry or something similar to prevent that